sonar-db/src/main/resources/org/sonar/db/ce/CeQueueMapper.xml

```diff
@@ -29,8 +29,13 @@
     where cq.uuid=#{uuid}
   </select>
 
-  <select id="countByStatus" parameterType="org.sonar.db.ce.CeQueueDto$Status" resultType="int">
-    select count(1) from ce_queue where status=#{status}
+  <select id="countByStatusAndComponentUuid" parameterType="map" resultType="int">
+    select count(1)
+    from ce_queue
+    where status=#{status}
```

Comment:
after discussion, I will NOT add an index on CE_QUEUE_STATUS for several reasons:

- for an index to be efficient, the filter must represent less than 10% of the table (our nominal case filter on 'PENDING' status)
- an index slows insert/update/delete operations and it's critical to have fast inserts on the CE_QUEUE table
- in general, we don't add an index for small table. I know the table could potentially be huge, but that's a sign of a problem. The table size should tend towards zero. Anyway, as stated, adding an index would help in this case